File: Easy/Java/1588. Sum of All Odd Length Subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int sum = 0;

        // Count each element in an odd-length sub-array and multiply
        // the array elements to get the output of the sum
        for(int i = 0; i < arr.length; i++){
            int present = ((i + 1) * (arr.length - i) + 1) / 2;
            sum += present * arr[i];
        }

        return sum;
    }
}

File: Easy/Python/1588. Sum of All Odd Length Subarrays.py
class Solution(object):
    def sumOddLengthSubarrays(self, arr):
        sum = 0

        # Count each element in an odd-length sub-array and multiply
        # the array elements to get the output of the sum
        for i in range(len(arr)):
            present = ((i + 1) * (len(arr) - i) + 1) / 2
            sum = sum + (present * arr[i])

        return sum
